File: Java/Data Structures/Trees/BinaryTrees/Binary-Tree-Postorder-Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer>values;
    public Solution(){
        this.values = new ArrayList<>();
    }
    public List<Integer> postorderTraversal(TreeNode root) {
        if(root==null){
            return this.values;
        }
        this.postorderTraversal(root.left);
        this.postorderTraversal(root.right);
        if(root!=null){
            this.values.add(root.val);
        }
        return this.values;
    }
}

File: C++/Data Structures/Trees/BinaryTrees/Binary-Tree-Postorder-Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int>values;
    vector<int> postorderTraversal(TreeNode* root) {
        if(root==nullptr){
            return values;
        }
        postorderTraversal(root->left);
        postorderTraversal(root->right);
        if(root!=nullptr){
            values.push_back(root->val);
        }
        return values;
    }
};
